Mathlib/RingTheory/Derivation/MapCoeffs.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.RingTheory.Derivation.Basic
+import Mathlib.Algebra.Polynomial.Module.Basic
+
+/-!
+# Derivations of univariate polynomials
+
+In this file we prove properties about `R`-derivations of `Polynomial A`.
+-/
+
+noncomputable section
+
+open Polynomial Module
+
+namespace Derivation
+
+variable {R A M : Type*} [CommRing R] [CommRing A] [Algebra R A] [AddCommGroup M]
+  [Module A M] [Module R M] [IsScalarTower R A M] (d : Derivation R A M) (a : A)
+
+/--
+The `R`-derivation from `A[X]` to `M[X]` which applies the derivative to each
+of the coefficients.
+-/
+def mapCoeffs : Derivation R A[X] (PolynomialModule A M) where
+  __ := (PolynomialModule.map A d.toLinearMap).comp
+    PolynomialModule.equivPolynomial.symm.toLinearMap
+  map_one_eq_zero' := show (Finsupp.single 0 1).mapRange (d : A → M) d.map_zero = 0 by simp
+  leibniz' p q := by
+    dsimp
+    induction p using Polynomial.induction_on' with
+    | h_add => simp only [add_mul, map_add, add_smul, smul_add, add_add_add_comm, *]
+    | h_monomial n a =>
+      induction q using Polynomial.induction_on' with
+      | h_add => simp only [mul_add, map_add, add_smul, smul_add, add_add_add_comm, *]
+      | h_monomial m b =>
+        refine Finsupp.ext fun i ↦ ?_
+        dsimp [PolynomialModule.equivPolynomial, PolynomialModule.map]
+        simp only [toFinsupp_mul, toFinsupp_monomial, AddMonoidAlgebra.single_mul_single]
+        show d _ = _ + _
+        erw [Finsupp.mapRange.linearMap_apply, Finsupp.mapRange.linearMap_apply]
+        rw [Finsupp.mapRange_single, Finsupp.mapRange_single]
+        erw [PolynomialModule.monomial_smul_single, PolynomialModule.monomial_smul_single]
+        simp only [AddMonoidAlgebra.single_apply, apply_ite d, leibniz, map_zero, coeFn_coe,
+          PolynomialModule.single_apply, ite_add_zero, add_comm m n]
+
+@[simp]
+lemma mapCoeffs_apply (p : A[X]) (i) :
+    d.mapCoeffs p i = d (coeff p i) := rfl
+
+@[simp]
+lemma mapCoeffs_monomial (n : ℕ) (x : A) :
+    d.mapCoeffs (monomial n x) = .single A n (d x) := Finsupp.ext fun _ ↦ by
+  simp [coeff_monomial, apply_ite d, PolynomialModule.single_apply]
+
+@[simp]
+lemma mapCoeffs_X :
+    d.mapCoeffs (X : A[X]) = 0 := by simp [← monomial_one_one_eq_X]
+
+@[simp]
+lemma mapCoeffs_C (x : A) :
+    d.mapCoeffs (C x) = .single A 0 (d x) := by simp [← monomial_zero_left]
+
+variable {K M' : Type*} [CommRing K] [Algebra R K] [Algebra A K]
```

Comment:
Could you please rename `K` to `B`? To me, `K` suggests that it is a field.

Note that, in VSCode, if you place your cursor on `K` and hit <F2> then you can rename it once, and it will update all other occurences of `K` for you.